From e13a3f91c56f6654c59bf9e50280f7a331fcfeec Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:54 -0400
Subject: [tools] perf: Fix memory leak in event_format__print function

Message-id: <1410181752-25631-50-git-send-email-jolsa@redhat.com>
Patchwork-id: 91731
O-Subject: [PATCH RHEL7.1 BZ1134356 049/307] perf tools: Fix memory leak in event_format__print function
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit b58f608e31010cb76ee953a6919f9d96b4eb58d9
Author: Jiri Olsa <jolsa@redhat.com>
Date: Sun Feb 2 22:38:49 2014 +0100

description
===========
Properly destroying trace_seq object.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/trace-event-parse.c b/tools/perf/util/trace-event-parse.c
index e0d6d07..c36636f 100644
--- a/tools/perf/util/trace-event-parse.c
+++ b/tools/perf/util/trace-event-parse.c
@@ -126,6 +126,7 @@ void event_format__print(struct event_format *event,
  trace_seq_init(&s);
  pevent_event_info(&s, event, &record);
  trace_seq_do_printf(&s);
+ trace_seq_destroy(&s);
 }
 
 void parse_proc_kallsyms(struct pevent *pevent,
-- 
1.7.1